src/main/java/org/optaplanner/springboottaskassigning/solver/SolverTask.java

```diff
@@ -16,45 +16,39 @@
 
 package org.optaplanner.springboottaskassigning.solver;
 
-import java.util.function.Consumer;
+import java.util.concurrent.Callable;
 
 import org.optaplanner.core.api.score.Score;
 import org.optaplanner.core.api.solver.Solver;
 import org.optaplanner.core.api.solver.event.SolverEventListener;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class SolverTask<Solution_> implements Runnable {
+public class SolverTask<Solution_> implements Callable<Solution_> {
```

Comment:
Not sure what to think about the `Callable` interface here. The instance is not passed to `ExecutorService` - the `CompletableFuture.supplyAsync()` is called, taking a method reference to the `call` method directly.

Reply:
Good catch, there's no need to implement `Callable` anymore. Even better `SolverTask.call()` can be renamed to `SolverTask.startSolving()` which makes more sense.